core/src/Telemetry/DefaultEvent.cs

```diff
@@ -25,18 +25,35 @@
 //
 //------------------------------------------------------------------------------
 
-using Microsoft.Identity.Core;
+using System.Collections.Concurrent;
+using System.Globalization;
 
 namespace Microsoft.Identity.Core.Telemetry
 {
     internal class DefaultEvent : EventBase
     {
-        public DefaultEvent(string clientId) : base(EventNamePrefix + "default_event")
+        public DefaultEvent(string clientId, ConcurrentDictionary<string, int> eventCount) : base((string) (EventNamePrefix + "default_event"))
```

Comment:
You are passing in a concurrent dictionary yet you are using it like a normal vanilla dictionary. ConcurrentDictionary needs to be used with TryGet / AddOrUpdate etc. type of methods to ensure concurrency. AFAIK using normal dictionary methods on a ConcurrentDictionary defeats the concurrency insurances.

This being said, all C# collections are thread safe when it comes to reading. So if at the time of reading you're done with writing, you can use IReadonlyDictionary

Reply:
Good point. ConcurrentDictionary does not implement IReadOnlyDictionary until .NET framework 4.6.2. so made it IDictionary instead